File: iOS/RaiseUserEvent-MagicApp/RaiseUserEvent.m
//
//  RaiseUserEvent.m
//  MagicApp
//

#import "RaiseUserEvent.h"
#import "Magicxpa.h"

@interface RaiseUserEvent ()

@end

@implementation RaiseUserEvent

+(void)raiseEvent:(NSArray *)params
{
	// raise a user event
	// first argument is the event name
	// second..n arguments are the parameters
	NSString *eventName = [params objectAtIndex: 0];
	NSString *msg1 = [params objectAtIndex: 1];
	NSString *msg2 = [params objectAtIndex: 2];

    NSArray *paramstosend = [NSArray arrayWithObjects:msg1, msg2, nil];
    [Magicxpa invokeUserEvent:eventName Params:paramstosend];
}

@end

File: Android/RaiseUserEvent-src/RaiseUserEvent.java
package com.magicsoftware.magicdev;

import com.magicsoftware.core.CoreApplication;

public class RaiseUserEvent {
	// raise a user event
	// first argument is the event name
	// second..n arguments are the parameters
	public static void raiseEvent(String eventName, String msg1, String msg2) {
		CoreApplication.getInstance().invokeUserEvent(eventName,msg1,msg2); 
	}
}
